proto/spacemesh/types.proto

```diff
@@ -0,0 +1,162 @@
+syntax = "proto3";
+package spacemesh;
+
+message SimpleInt {
+    uint64 value = 1;
+}
+
+message SimpleString {
+    string value = 1;
+}
+
+// a non-negative coin amount
+message Amount {
+    uint64 value = 1;
+}
+
+message AccountId {
+    bytes address = 1;
+}
+
+message Account {
+    AccountId address = 1;  // account public address
+    bytes public_key = 2;   // nice to include the account full public key
+    uint64 counter = 3;     // aka nonce
+    Amount balance = 4;     // known account balance
+}
+
+enum TransactionProcessingStatus {
+    UNDEFINED = 0;
+    REJECTED = 1;               // rejected pre STF processing. e.g. bad data
+    INSUFFICIENT_FUNDS = 2;     // rejected pre STF processing by funds check.
+    CONFLICTING = 3;            // rejected pre STF due to conflicting counter.
+    PENDING = 4;                // pending processing by STF
+    //PROCESSING = 5;
+    //APPROVED = 6;
+    PROCESSED = 7;              // processed. Results are in the tx receipt.
+}
+
+enum TransactionType {
+    COIN = 0;            // simple coin transaction
+    ATX = 1;             // smesher activation transaction
+    APP = 2;             // smart contract method
+    DEPLOY = 3;          // deploy app from templlate
+    DEPLOY_TEMPLATE = 4; // deploy app template code to mesh
+}
+
+enum TransactionResult { // results of processing a transaction by the STF
+    RUNTIME_EXCEPTION = 0;              // app code exception
+    INSUFFICIENT_GAS_PROVIDED = 1;      // out of gas
+    INSUFFICIENT_BALANCE = 2;           // failed due to sender insufficient funds
+    EXECUTED = 3;                      // executed w/o error
+}
+
+message TransactionReceipt {
+    TransactionId id = 1;  // the source transaction
+    TransactionResult result = 2;   // tx processing result
+    uint64 gas_used = 3;            // gas units used by the transaction (gas price in tx)
+    Amount fee = 4;                 // transaction fee charged for the transaction
```

Comment:
How is fee different from gas_used?

Reply:
gas_used is in gas units. Fee is the fee in smidge which is equal to gas_used * gas_price set by transaction caller.

Reply:
I think it might be more intuitive if, instead of the receipt containing `gas_used` and `fee`, it instead contained `gas_used` and `gas_price`, like the tx itself. The math is the same.